cns/restserver/internalapi.go

```diff
@@ -475,6 +475,25 @@ func (service *HTTPRestService) DeleteNetworkContainerInternal(
 	return types.Success
 }
 
+func (service *HTTPRestService) EnsureNoStaleNCs(validNCIDs []string) {
+	valid := make(map[string]struct{})
+	for _, ncID := range validNCIDs {
+		valid[ncID] = struct{}{}
+	}
+
+	service.Lock()
+	defer service.Unlock()
+
+	for ncID := range service.state.ContainerStatus {
```

Comment:
Lets also make sure there is no state in the cache data. like PodIPMap, and if there is then it is not assigned to any pod.

Reply:
is this possible? we won't be able to rebuild the cache with stale data since 'service.state.ContainerStatus' doesn't have the stale NC anymore, no?